From: Egbert Eich <eich@suse.de>
Date: Sat Aug 22 11:42:26 2015 +0200
Subject: drm/i915: WARN_ONCE() about unexpected interrupts for all chipsets
Patch-mainline: N/A
References: bsc#942938

With
drm/i915: Avoid race of intel_crt_detect_hotplug() with HPD interrupt
we can remove:

commit 3ff04a160a891e56cdcee5c198d4c764d1c8c78b
Author: Daniel Vetter <daniel.vetter@ffwll.ch>
Date:   Thu Apr 24 12:03:17 2014 +0200

    drm/i915: Don't WARN nor handle unexpected hpd interrupts on gmch platforms

Signed-off-by: Egbert Eich <eich@suse.de>
Signed-off-by: Egbert Eich <eich@suse.com>
---
 drivers/gpu/drm/i915/i915_irq.c | 17 ++++-------------
 1 file changed, 4 insertions(+), 13 deletions(-)
diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 928ebf9..4b89e39 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -666,19 +666,10 @@ static inline void intel_hpd_irq_handler(struct drm_device *dev,
 	spin_lock_irqsave(&dev_priv->irq_lock, irqflags);
 	for (i = 1; i < HPD_NUM_PINS; i++) {
 
-		if (hpd[i] & hotplug_trigger &&
-		    dev_priv->hpd_stats[i].hpd_mark == HPD_DISABLED) {
-			/*
-			 * On GMCH platforms the interrupt mask bits only
-			 * prevent irq generation, not the setting of the
-			 * hotplug bits itself. So only WARN about unexpected
-			 * interrupts on saner platforms.
-			 */
-			WARN_ONCE(INTEL_INFO(dev)->gen >= 5 && !IS_VALLEYVIEW(dev),
-				  "Received HPD interrupt (0x%08x) on pin %d (0x%08x) although disabled\n",
-				  hotplug_trigger, i, hpd[i]);
-			continue;
-		}
+		WARN_ONCE((hpd[i] & hotplug_trigger) &&
+			  (dev_priv->hpd_stats[i].hpd_mark != HPD_ENABLED),
+			  "Received HPD interrupt (0x%08x) on pin %d (0x%08x) although disabled\n",
+			  hotplug_trigger, i, hpd[i]);
 
 		if (!(hpd[i] & hotplug_trigger) ||
 		    dev_priv->hpd_stats[i].hpd_mark != HPD_ENABLED)
